src/main/java/rodain/rules/filters/FilterIgnored.java

```diff
@@ -0,0 +1,48 @@
+package rodain.rules.filters;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+/**
+ * Created by adrapereira on 20-10-2015.
+ */
+public class FilterIgnored implements ContentFilter {
+    private HashSet<String> ignored;
+
+    public FilterIgnored(){
+        ignored = new HashSet<>();
+    }
+
+    @Override
+    public void add(String st){
+        ignored.add(st);
+    }
+
+    @Override
+    public void addAll(Collection col){
+        ignored.addAll(col);
+    }
+
+    @Override
+    public boolean filter(String st){
+        if(ignored.contains(st))
+            return true;
+        int index = 0, end = st.length(), fromIndex = 0;
+        while(index < end && (index = st.indexOf("/", fromIndex)) != -1){
```

Comment:
This method is very difficult to understand. Needs javadoc and comments at the least.
Also, it is not nice to change the variables that are being evaluated on the evaluating expression itself. Please make the code more simple, even if using more lines of code.